src/telemetry/index.ts

```diff
@@ -8,19 +8,54 @@ export class TelemetryService implements vscode.Disposable {
   private telemetryReporter: TelemetryReporter = new TelemetryReporter(
     "50598369-dd83-4f9a-9a65-ca1fa6f1785c",
   );
+  private eventMeasurements = new Map();
 
   setTelemetryCustomAttribute(key: string, value: string) {
     this.customAttributes[key] = value;
   }
 
-  sendTelemetryEvent(
+  startTelemetryEvent(
     eventName: string,
     properties?: { [key: string]: string },
     measurements?: { [key: string]: number },
+  ) {
+    this.eventMeasurements.set(eventName, new Date().getTime());
+    this.sendTelemetryEvent(eventName, properties, measurements);
+  }
+
+  // TODO: check if we have to identify python exception
+  endTelemetryEvent(
+    eventName: string,
+    error?: unknown,
+    properties?: { [key: string]: string },
+    measurements?: { [key: string]: number },
+  ) {
+    const start = this.eventMeasurements.get(eventName);
+    if (error) {
+      this.sendTelemetryError(`${eventName}Error`, error, properties, {
+        ...(measurements || {}),
+        duration: new Date().getTime() - start,
+      });
+    }
+  }
+
+  private getFeatureName(eventName: string) {
+    const [featureName, rest] = eventName.split("/");
+    if (rest) {
+      return { feature: featureName };
+    }
+    return {};
+  }
+
+  sendTelemetryEvent(
+    eventName: string, // TODO: should be TelemetryEvents
```

Comment:
There is an open TODO here.

Reply:
This is to be addressed after we migrate all telemetry events to this new format